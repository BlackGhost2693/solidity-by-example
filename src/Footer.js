import React, { Component } from "react"
import "./Footer.css"

class Footer extends Component {
  render() {
    return (
      <p className="footer">
        <a href="mailto:soliditybyexample@gmail.com">
          soliditybyexample@gmail.com
        </a>{" "}
        |{" "}
        <a href="https://github.com/solidity-by-example/solidity-by-example.github.io">
          source
        </a>{" "}
        |{" "}
        <a href="https://github.com/solidity-by-example/solidity-by-example.github.io/tree/gh-pages#license">
          license
        </a>
      </p>
    )
  }
}

export default Footer
