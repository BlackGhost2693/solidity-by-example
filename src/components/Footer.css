p.footer {
    color: grey;
}
p.footer a, p.footer a:visited {
    color: grey;
}
