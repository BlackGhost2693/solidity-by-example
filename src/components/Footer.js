import React, { Component } from "react"
import styles from "./Footer.module.css"

function Footer() {
  return (
    <div className={styles.component}>
      <div className={styles.row}>
        <a
          href="https://etherscan.io/address/0xA8DD4bA80d8904d7cAcE5c3622CCf78bEa5AcFEe"
          target="__blank"
        >
          Donate: 0xA8DD4bA80d8904d7cAcE5c3622CCf78bEa5AcFEe
        </a>
      </div>
      <div className={styles.row}>
        <a href="mailto:contact@solidity-by-example.org">
          contact@solidity-by-example.org
        </a>
        <div className={styles.bar}>|</div>
        <a
          href="https://github.com/solidity-by-example/solidity-by-example.github.io"
          target="__blank"
        >
          source
        </a>
        <div className={styles.bar}>|</div>
        <a
          href="https://github.com/solidity-by-example/solidity-by-example.github.io/blob/gh-pages/LICENSE"
          target="__blank"
        >
          license
        </a>
      </div>
    </div>
  )
}

export default Footer
