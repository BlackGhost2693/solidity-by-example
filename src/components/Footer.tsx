import React from "react"
import styles from "./Footer.module.css"

// Khan Academy
const ADDRESS = "0x95a647B3d8a3F11176BAdB799b9499C671fa243a"

function Footer() {
  return (
    <div className={styles.component}>
      <div className={styles.row}>
        Donation goes directly to
        <a
          href="https://khanacademy.org"
          target="__blank"
          className={styles.khan}
        >
          Khan Academy
        </a>
        <span role="img" aria-label="smiley">
          ðŸ™‚
        </span>
      </div>
      <div className={styles.row}>
        <a href={`https://etherscan.io/address/${ADDRESS}`} target="__blank">
          {ADDRESS.slice(0, 6)}...{ADDRESS.slice(-6, -1)}
        </a>
      </div>
      <div className={styles.row}>
        <a href="mailto:contact@solidity-by-example.org">
          contact@solidity-by-example.org
        </a>
      </div>
      <div className={styles.row}>
        <a
          href="https://github.com/solidity-by-example/solidity-by-example.github.io"
          target="__blank"
        >
          source
        </a>
        <div className={styles.bar}>|</div>
        <a
          href="https://github.com/solidity-by-example/solidity-by-example.github.io/blob/gh-pages/LICENSE"
          target="__blank"
        >
          license
        </a>
      </div>
    </div>
  )
}

export default Footer
