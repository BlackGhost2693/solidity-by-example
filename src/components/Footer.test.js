import React from "react"
import { shallow } from "enzyme"
import Footer from "./Footer"

test("it renders", () => {
  const component = shallow(<Footer />)

  expect(component).toMatchSnapshot()
})
