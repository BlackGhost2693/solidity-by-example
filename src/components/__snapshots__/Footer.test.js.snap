// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<p
  className="footer"
>
  <a
    href="mailto:contact@solidity-by-example.org"
  >
    contact@solidity-by-example.org
  </a>
   
  |
   
  <a
    href="https://github.com/solidity-by-example/solidity-by-example.github.io"
  >
    source
  </a>
   
  |
   
  <a
    href="https://github.com/solidity-by-example/solidity-by-example.github.io/tree/gh-pages#license"
  >
    license
  </a>
</p>
`;
